pkg/util/log/stderr_redirect_unix.go

```diff
@@ -18,13 +18,19 @@ package log
 
 import (
 	"os"
+	"syscall"
 
 	"golang.org/x/sys/unix"
 )
 
 func dupFD(fd uintptr) (uintptr, error) {
-	nfd, err := unix.Dup(int(fd))
-	return uintptr(nfd), err
+	// Warning: failing to set FD_CLOEXEC on duplicated file descriptors can
+	// prevent the process from exiting while child processes are live.
```

Comment:
Can you expand on this? I think I'm about to learn something, but so far I'm just confused as to why a file descriptor in the child process can keep the parent process from exiting.

Reply:
I'm also interested in an explanation. @bdarnell wrote earlier,

> when it kills the subprocess, it reads its stdout and stderr to eof. if that file descriptor was passed into a child process, i think it would wait for the child process to complete

but I'm not clear on the details at all.